### ISSUE TITLE
Found nothing to obfuscate

### ISSUE THREAD

**Issue by luanlzsn** [rockbruno/swiftshield#39]
I use swiftshield -automatic，Found nothing to obfuscate. Do I need to do anything about it?

**Comment by rockbruno** [comment]
I usually get that when I mistype something, are you sure everything is correct?

```
cd ./myProjectRoot
./swiftshield -automatic -project-root ./ -automatic-project-scheme MyScheme -automatic-project-file ./myProject.xcworkspace
```

**Comment by luanlzsn** [comment]
<img width="570" alt="2018-12-15 19 43 55" src="https://user-images.githubusercontent.com/7351125/50042710-d2548580-00a1-11e9-8b27-7f0edbf78ad3.png">
Still the same

**Comment by luanlzsn** [comment]
Thanks，Really caused by the scheme's name，I changed my scheme's name to Tocos and I succeeded.

**Comment by rockbruno** [comment]
That’s bad! You should be able to run this with any scheme name. I’ll investigate it and push a fix.

**Comment by rockbruno** [comment]
By the way, I added an `-` to the example project's scheme and it worked fine 😢 
If you don't mind, can you upload to me a version of your code where that error happened? I had it happen to me a few times and it always went away, so I wonder if it's something on my code or SourceKit.

**Comment by rockbruno** [comment]
Thanks! Going to take a look.

**Comment by rockbruno** [comment]
Thanks @luanlzsn !
I actually had no problem with the scheme name (changed it to `Boxy` and it started obfuscating just fine, so I'm not sure why that happened to you), but I was also unable to compile the project just like you.

The problem is that you've found a SourceKit issue that I hadn't seen before, in which SourceKit fails to index things that are marked as unavailable:

```
@available(*, unavailable, renamed: "ResultProtocol")
public typealias ResultType = ResultProtocol
```

In your project, that affects NVActivityIndicatorView, RxSwift, R.swift and Result as far as I could tell. If you ignore obfuscating them, your project should be able to compile.

I'm going to add this to [SOURCEKITISSUES.md](https://github.com/rockbruno/swiftshield/blob/master/SOURCEKITISSUES.md) and work on it on the Swift repo when I get some time in the next few days.

**Comment by git-jkumar** [comment]
Hi,
I am getting same error. I am not sure how installation will work.

./bin/swiftshield -project-root ./ -automatic -automatic-project-file ./SwiftShieldExample.xcworkspace -automatic-project-scheme SwiftShieldExample
SwiftShield 3.3.1
Automatic mode
Building project to gather modules and compiler arguments...
-- Searching for references of the retrieved USRs --
Found nothing to obfuscate. Finishing...

**Comment by rockbruno** [comment]
Hey @git-jkumar , does this happens if you CD to the example folder and use the bash file available there too?

**Comment by git-jkumar** [comment]
Yes. I tried the same but no result. Please see below result
/Users/user/Documents/swiftshieldMaa/SwiftShieldExample/_CDToMeAndRunMeToObfuscate.sh 
SwiftShield 3.3.1
Automatic mode
Building project to gather modules and compiler arguments...
-- Searching for references of the retrieved USRs --
Found nothing to obfuscate. Finishing...

**Comment by patriRich** [comment]
Try installing the command line tools for your Xcode version, and then point to the correct version of the xcodebuild using the next command:
sudo xcode-select -s /Applications/Xcode.app/Contents/Developer

Then, the script (and the tool) should run properly.

**Comment by zfoot** [comment]
Hi, I tried your obfuscating tools, after fixing the above issues, swiftshield did obfuscate however it only obfuscate my pod project source code(mainly RxSwift) and my source code is untouched. I'm running the command in automatic mode:

./swiftshield -automatic -project-root ./ -automatic-project-scheme MyProject -automatic-project-file ./MyProject.xcworkspace

**Comment by PeterHdd** [comment]
Im trying to obsucate an ionic project. I went to the directory of the ios app and did the following:

`swiftshield -automatic -project-root ./ -automatic-project-scheme ProjectName -automatic-project-file ./ProjectName.xcworkspace`

But I keep getting the same error:

>Searching for references of the retrieved USRs

>Found nothing to obfuscate. Finishing...

Can anyone help?

**Comment by rockbruno** [comment]
I'm not sure how Ionic projects works, do they generate full native swift files?

**Comment by PeterHdd** [comment]
yes, in ionic you execute the command:

`ionic cordova platform add ios` then `ionic cordova build ios`  and then you can open the ios app in xcode.

But all the classes, interfaces.. are in objective c and not swift.

**Comment by rockbruno** [comment]
So no Swift files? That could be the problem since automatic mode only works for Swift classes.

**Comment by PeterHdd** [comment]
The manual mode did create a file called `conversionMap.txt` as said in the docs. But it only has this:

__s ===> 0BHaa.....

So im not sure does this mean it obfuscated the whole app?

**Comment by ligun123** [comment]
I found that if the scheme is different from the older, the script also failed.
Then I changed the schema as same as the older, the script worked.

**Comment by PeterHdd** [comment]
Since this plugin didnt work with ionic, I tried another plugin which also didnt work.

Then I tried to add `#define ClassName as1`, but that also didnt work because ionic was not apple to map the classes and methods written in the `config.xml` of each plugin to the objective c classes.

So, I ended up downloading each plugin and changing the class names and methods and then adding the plugin in ionic :/

**Comment by YunyueLin** [comment]
hi, @rockbruno

**Comment by YunyueLin** [comment]
hi, @rockbruno  
I can't run the SwiftShieldExample , and get same error.
- use _CDToMeAndRunMeToObfuscate.sh
- use command code 
My Xcode version and command line tools is 10.2 . 
What should I do ? 
![企业微信截图_0453014b-1d37-441d-88f7-db2c830dce56](https://user-images.githubusercontent.com/12877953/56030589-ca56da80-5d4f-11e9-97e3-a80061061653.png)

**Comment by rockbruno** [comment]
The Xcode I used to make that project is 10.1, I'll update it to the latest one which might treat these issues.

**Comment by YunyueLin** [comment]
> The Xcode I used to make that project is 10.1, I'll update it to the latest one which might treat these issues.

ha. thank you.

**Comment by rockbruno** [comment]
Yup I was able to reproduce it, it looks like swiftshield doesn't work very well with Xcode's new build mode... I was able to make it run successfully by changing the workspace back to legacy mode.
I've pushed these changes, can you see if it works now?

**Comment by YunyueLin** [comment]
It's work, great!

**Comment by rockbruno** [comment]
Alright, so I investigated this further... The reason it wasn't finding anything to obfuscate is because I need to slightly change the build parameters when running projects with the new build system.

The problem is that it looks like they changed how indexing works in this new Xcode. I can run the tool normally, but all references from other modules don't get obfuscated. They don't even show up in the logs anymore... Trying to discover what changed...

**Comment by tomtom13** [comment]
@rockbruno, what have you modified in your project to actually make it work ?

**Comment by rockbruno** [comment]
@tomtom13 If you mean what the fix for the example project was, all you need to do is change your workspace settings (under File) to use Legacy Build:

![Uploading Screen Shot 2019-04-13 at 5.11.02 PM.png…]()

To make it work with non-legacy builds I need to push a new version of the tool, but Swift 5 isn't very friendly with it which is what I'm trying to solve right now.

**Comment by tomtom13** [comment]
Sorry mate but the screen shot doesn't load. 

Anyway I've went into File -> Workspace settings
Changed both build settings to legacy
Then after hitting advanced I changed paths to legacy ..

This "Found nothing to obfuscate. Finishing..."

**Comment by tomtom13** [comment]
Look, I don't want to rock in with the swag saying "fix it" :) so if you need anything from me - let me know.

**Comment by rockbruno** [comment]
No worries! 99% of the times this problem happens is because xcodebuild failed, and I just realize that this must be manually checked... I always thought that NSTask would automatically kill SwiftShield as well if it failed.

I've just pushed 3.3.2 now that hopefully fixes the "nothing to obfuscate" problem when using the new build system, and SwiftShield will now finally print the xcodebuild logs if it fails, so hopefully the "nothing to obfuscate" will never happen anymore - you should now get exactly what went wrong with Xcode.

**Comment by tomtom13** [comment]
Thanks - now there is a meaningfull output. I get :
"The "-list" option can be used to find the names of the schemes in the workspace."

Which is surprising because I tried using app name, I tried using "archive" "run" etc.

**Comment by AleksanderDeveloper** [comment]
[
![Снимок экрана 2020-04-21 в 11 56 54](https://user-images.githubusercontent.com/51282811/79846523-39392b80-83c7-11ea-827e-915785a595ff.png)
](url)

**Comment by AleksanderDeveloper** [comment]
Brothers))) Help with the problem, how do I use swiftshield? What am I doing wrong ?

**Comment by AleksanderDeveloper** [comment]
In general, do I need to install the co-apods pod 'SwiftShield', '~> 3.5'? into the app?

**Comment by AleksanderDeveloper** [comment]
can you explain step by step how to use me Swiftshield ?

**Comment by AleksanderDeveloper** [comment]
<img width="803" alt="Снимок экрана 2020-04-21 в 12 03 23" src="https://user-images.githubusercontent.com/51282811/79847344-4571b880-83c8-11ea-937e-da6bf53e765f.png">

<img width="786" alt="Снимок экрана 2020-04-21 в 12 05 14" src="https://user-images.githubusercontent.com/51282811/79847429-65a17780-83c8-11ea-9cf4-7dc7fda57063.png">

<img width="758" alt="Снимок экрана 2020-04-21 в 12 05 22" src="https://user-images.githubusercontent.com/51282811/79847453-6cc88580-83c8-11ea-830b-2ab6b4019aa2.png">

Is it considered that I installed swiftshield?

**Comment by AleksanderDeveloper** [comment]
please help please! I ask for help please! I ask for help please! I ask for help please! I ask for help please! I ask for help please! I ask for help please! I ask for help please! I ask for help please! I ask for help please! I ask for help please! I ask for help please! I ask for help please! And then they will kick me out of work))))

**Comment by tomtom13** [comment]
I don't think you will get any help with that kind of attitude. And if you don't know something that your work depends on and can get you fired within hours - you should be kicked out of work because you don't belong there.

**Comment by rockbruno** [comment]
Alexsander, I see that you are trying to copy it to /usr/bin, which probably isn't a good idea as this tool relies on the path that it gets executed on. The README has all the info you need. You shouldn't have problems if you follow it step by step.

**Comment by AleksanderDeveloper** [comment]
<img width="638" alt="Снимок экрана 2020-04-22 в 10 34 04" src="https://user-images.githubusercontent.com/51282811/79953637-dad28280-8484-11ea-85a6-59bdb4174029.png"> 

Hello!) Do I need to install these POD ??

**Comment by AleksanderDeveloper** [comment]
![Снимок экрана 2020-04-22 в 10 35 46](https://user-images.githubusercontent.com/51282811/79953758-09e8f400-8485-11ea-9880-5c86a11ebb15.png)

What does this error mean?

**Comment by rockbruno** [comment]
Again, please read the README. CocoaPods is never mentioned as an installation method and so is unsupported.
